js/ad_manager_ooyala.js

```diff
@@ -0,0 +1,207 @@
+     * ready to be parsed.
+     * This metadata may contain the adTagUrl and other ad manager and movie specific configuration.
+     * @method AdManager#loadMetadata
+     * @public
+     * @param {object} adManagerMetadata Ad manager-specific metadata
+     * @param {object} backlotBaseMetadata Base metadata from Ooyala Backlot
+     * @param {object} movieMetadata Metadata for the main video
+     */
+    this.loadMetadata = function(adManagerMetadata, backlotBaseMetadata, movieMetadata) {
+      if (!movieMetadata) return;
+      this.movieMetadata = movieMetadata;
+      if (_getMainVideoEmbedCode == true) {
+        _movieEmbedCode = movieMetadata.embed_code;
+        _getMainVideoEmbedCode = false;
+      }
+      _adEmbedCode = movieMetadata.ads[0].ad_embed_code;
+      var adsRequest = {
+        pcode: movieMetadata.asset_pcode || "unknown",
+        embedCode: movieMetadata.ads[0].ad_embed_code,
+        server: OO.SERVER.AUTH,
+        params: {}
+      };
+      _amc.notify(_amc.EVENTS.WILL_FETCH_AD_AUTHORIZATION,adsRequest); 
+    };
+
+    /**
+     * Called by Ad Manager Controller.  When this function is called, all the information about the Ooyala
+     * ads will be received and will notify the controller that the Ad Manager is ready.
+     * @public
+     * @param {object} adData Ad-specific metadata
+     */
+    this.ooyalaAdData = function(adData) {
+      if (!adData) return;
+      this.streams[OO.VIDEO.ENCODING.MP4] = adData;
+      this.ready = true;
+      _amc.onAdManagerReady();
+    };
+
+    /**
+     * Called once per video by Ad Manager Controller once the ad manager has set its ready flag to true.
+     * This function asks the ad manager to return a list of all ads to the controller for addition in the
+     * line. If the list of ads is not available at this time, return [] or null and call
+     * [AdManagerController].appendToTimeline() when the ads become available.
+     * The duration and position of each ad should be specified in seconds.
+     * @method AdManager#buildTimeline
+     * @public
+     * @returns {OO.AdManagerController#Ad[]} timeline A list of the ads to play for the current video
+     */
+    this.buildTimeline = function() {
+      var adsTimeline = [];
+      var adData = {
+        "position": this.movieMetadata.ads[0].time,
+        "adManager": this.name,
+        "ad": this.movieMetadata.ads[0],
+        "streams": this.streams,
+        "adType": _amc.ADTYPE.UNKNOWN_AD_REQUEST
+      };
+      var adToInsert = new _amc.Ad(adData);
+      adsTimeline.push(adToInsert);  
+      return adsTimeline;       
+    };
+
+    /**
+     * Called by Ad Manager Controller. The ad manager should play the ad or group of podded ads passed to
+     * the function as a parameter.
+     * @method AdManager#playAd
+     * @public
+     * @param {object} ad The ad object to play
+     */
+    this.playAd = function(ad) {
+      var ooyalaAdAutoPlay = {"autoPlay" : true}; 
```

Comment:
Can we rename this variable to something that references that this is a params object? It will help with readability.